packages/lib/src/vis/shared/ViewportCenterer.tsx

```diff
@@ -17,9 +17,14 @@ function ViewportCenterer() {
   });
 
   useEffect(() => {
+    // On resize, move camera to the latest saved viewport center coordinates
     if (viewportCenter.current) {
-      // On resize, move camera to the latest saved viewport center coordinates
-      moveCameraTo(dataToWorld(viewportCenter.current));
+      const newPos = dataToWorld(viewportCenter.current);
+
+      // Ignore previous center if no longer valid with current scale
+      if (Number.isFinite(newPos.x) && Number.isFinite(newPos.y)) {
+        moveCameraTo(newPos);
+      }
```

Comment:
Subtle bug fix here: when switching from linear to log when visualising a dataset with a negative domain in the _Line_ vis, the viewport center saved (in data coordinates) while in linear scale cannot be converted back to world coordinates in log scale (because `ordinateScale(<negativeValue>)` returns `NaN`).

This breaks the visualization, because `ViewportCenterer` moves the camera position to a `NaN` y coordinate:

[Screencast from 2025-03-18 16-11-20.webm](https://github.com/user-attachments/assets/b8b2ca81-9b26-4d52-9a1f-3f50c0876ebc)

Reply:
:fishing_pole_and_fish: